python/lsst/obs/hsc/transformRegistry.py

```diff
@@ -247,20 +255,75 @@ class HscDistortionConfig(Config):
                                    -1.03493809035e-37,
                                    8.09745928121e-38,
                                    ])
-    tolerance = Field(dtype=float, default=5.0e-3, doc="Tolerance for inversion (pixels)") # Much less than 1
-    maxIter = Field(dtype=int, default=10, doc="Maximum iterations for inversion") # Usually sufficient
+    tolerance = Field(dtype=float, default=5.0e-3, doc="Tolerance for inversion (pixels)")  # Much less than 1
+    maxIter = Field(dtype=int, default=10, doc="Maximum iterations for inversion")  # Usually sufficient
     plateScale = Field(dtype=float, default=1.0, doc="Plate scale (arcsec/mm)")
 
 
+def makeAstPolyMapCoeffs(order, xCoeffs, yCoeffs):
+    """Convert polynomial coefficients in HSC format to AST PolyMap format
+
+    Paramaters
+    ----------
+    order: int
+        Polynomial order
+    xCoeffs, yCoeffs: list of float
+        Forward or inverse polynomial coefficients for the x and y axes
+        of output, in this order:
+            x0y0, x0y1, ...x0yN, x1y0, x1y1, ...x1yN-1, ...
+        where N is the polynomial order.
+
+    Returns
+    -------
+    Forward or inverse coefficients for astshim.PolyMap
+    as a 2-d numpy array.
+    """
+    nCoeffs = (order + 1) * (order + 2) // 2
+    if len(xCoeffs) != nCoeffs:
+        raise ValueError("found %s xCcdToSky params; need %s" % (len(xCoeffs), nCoeffs))
+    if len(yCoeffs) != nCoeffs:
+        raise ValueError("found %s yCcdToSky params; need %s" % (len(yCoeffs), nCoeffs))
+
+    coeffs = np.zeros([nCoeffs * 2, 4])
+    i = 0
+    for nx in range(order + 1):
+        for ny in range(order + 1 - nx):
+            coeffs[i] = [xCoeffs[i], 1, nx, ny]
+            coeffs[i + nCoeffs] = [yCoeffs[i], 2, nx, ny]
```

Comment:
What an awful convention for `coeffs`! I suggest that the "HSC format" makes sense, and there should be a helper function in astshim or afw to do the conversion.

Reply:
The convention may not be pretty, but it supports N->M axes. We already use 2->2, 2->1 and 1->1 in our stack, so such generality is useful. Let me think about it.

Reply:
I looked into a set of utility functions but decided it was best to leave this as it is for now. This order is reasonable, but swapping x and y axes would be equally valid. I'd rather not assume.